### PR TITLE
gsoc: [please merge :P ] LabHub: Add github plugin with invite feature

### PR DESCRIPTION
**Pull request by meetmangukiya**
# Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [x] Handle every possible exception.
- [x] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [x] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

Closes https://github.com/coala/corobo/issues/2

### CODE REVIEW

File: plugins/github.plug [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Core]
+module = github
+name = github
+
+[Documentation]
+description = GitHub utilities exposed through errbpt
```
Comment (meetmangukiya) [inline plugins/github.plug]: typo

File: plugins/github.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import logging
```
Comment (meetmangukiya) [inline plugins/github.py]: use internal logging object provided by errbot instead

File: plugins/github.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                "maintainers": "coala maintainers"
+            }
+
+            # Call the helper method after all the validation is done
+            Github.invite(inviter, self.TEAMS[team_mapping[team.lower()]])
+            return self.INVITE_SUCCESS.format(invitee=invitee)
+        else:
+            return "Nice try :poop:"
+
+    @re_botcmd(pattern=r"Hello\s*,?\s*world", flags=re.IGNORECASE)
```
Comment (meetmangukiya) [inline plugins/github.py]: re is not imported

File: plugins/github.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    def invite_hello_world(self, msg, match):
```
Comment (meetmangukiya) [inline plugins/github.py]: this should go in next commit

File: plugins/github.plug [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+name = github
+
+[Documentation]
+description = GitHub utilities exposed through errbot
+
+[Python]
+version = 3
+
+[Errbot]
+
```
Comment (meetmangukiya) [inline plugins/github.plug]: remove trailing line

File: plugins/labhub.py [hunk 6]
```diff
@@ -1,7 +1,106 @@
...
+                'newcomers': 'coala newcomers',
+                'developers': 'coala developers',
+                'maintainers': 'coala maintainers'
+            }
+
+            # Call the helper method after all the validation is done
+            LabHub.invite(invitee, self.TEAMS[team_mapping[team.lower()]])
+            return self.INVITE_SUCCESS[team.lower()].format(invitee)
+        else:
+            return 'Nice try :poop:'
```
Comment (Udayan12167) [inline plugins/labhub.py]: Better to say you are not a maintainer and hence you don't have the rights to invite. Nice try 💩 .

File: plugins/labhub.py [hunk 7]
```diff
@@ -1,7 +1,106 @@
...
+                     "\n\n*Do not take an issue if you don't understand it on "
+                     'your own.* Especially if you are new you have to be '
+                     'aware that getting started with an open source community '
+                     'is not trivial: you will have to work hard and most '
+                     'likely become a better coder than you are now just as we '
+                     "all did.\n\nDon't get us wrong: we are *very* glad to "
+                     'have you with us on this journey into open source! We '
+                     'will also be there for you at all times to help you with '
+                     'actual problems. :)',
+        'developers': 'Wow @{}, you are a part of developers team now! :tada: '
```
Comment (Udayan12167) [inline plugins/labhub.py]: I guess we should have the same explanation for what entails being a developer as well.

File: plugins/labhub.py [hunk 8]
```diff
@@ -104,3 +104,12 @@ def invite_cmd(self, msg, match):
...
         else:
             return 'Nice try :poop:'
+
+    def callback_message(self, msg):
+        """
+        Invite the user whose message includes the holy "hello world"
+        """
+        if re.search(r'hello\s*,?\s*world', msg.body, flags=re.IGNORECASE):
+            user = msg.frm.nick
+            LabHub.invite(user, self.TEAMS['coala newcomers'])
```
Comment (Udayan12167) [inline plugins/labhub.py]: Have a check here if already a member of a higher or same team.

File: plugins/labhub.py (hunk 7 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: But there is no developers corner.. there is maintainers corner :worried:

File: plugins/labhub.py (hunk 8 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: Nope, teams in github org are independent of one another, it doesn't lead to any bug.

However I should definitely check if the user is current member, would lead to reduction of reduntant invites when using hello world in a sentence :+1:

File: plugins/labhub.py [hunk 9]
```diff
@@ -1,7 +1,107 @@
...
 
+import github3
 
-class Labhub(BotPlugin):
+from errbot import BotPlugin, re_botcmd
+
+
+class LabHub(BotPlugin):
     """
     GitHub and GitLab utilities
```
Comment (jayvdb) [inline plugins/labhub.py]: unrelated, but eww ... https://github.com/coala/corobo/issues/101

File: plugins/labhub.py (hunk 7 above)
Comment (jayvdb) [inline plugins/labhub.py]: IMO as long as the messages are the same as cobot, we can merge and improve the usability during the project.

**Comment by jayvdb** [comment]
As a general comment, I would like to see all coala specific settings in a separate python module at least as CONSTANTS, not sprinkled through all the code.  And before the end of the project, these coala specific settings should all stored in a config file of some sort, not in the code.

E.g. the messages, and team names in this patch.

File: plugins/labhub.py (hunk 9 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: so, oneliners

File: plugins/labhub.py [hunk 10]
```diff
@@ -1,7 +1,104 @@
...
 
-class Labhub(BotPlugin):
-    """
-    GitHub and GitLab utilities
-    """
+from errbot import BotPlugin, re_botcmd
+
+
+class LabHub(BotPlugin):
+    '""GitHub and GitLab utilities""'
```
Comment (meetmangukiya) [inline plugins/labhub.py]: Buggy QuotesBear :worried:

File: plugins/labhub.py [hunk 11]
```diff
@@ -1,7 +1,104 @@
...
+from errbot import BotPlugin, re_botcmd
+
+
+class LabHub(BotPlugin):
+    """GitHub and GitLab utilities"""  # Ignore QuotesBear
+
+    def activate(self):
+        super().activate()
+
+        with open('plugins/LabHub.json') as f:
```
Comment (meetmangukiya) [inline plugins/labhub.py]: I am not sure this is gonna work :confused:

Comment (meetmangukiya) [inline plugins/labhub.py]: it'll work only if errbot is ran from project root

Comment (Udayan12167) [inline plugins/labhub.py]: So you can get the present module directory and use an absolute path?

Comment (meetmangukiya) [inline plugins/labhub.py]: yeah, maybe I should do that :+1:

Comment (meetmangukiya) [inline plugins/labhub.py]: I don't think there is a better solution then `os.path.join(os.environ.get('COBOT_ROOT', os.getcwd()), 'plugins/LabHub.json')`

File: plugins/labhub.py [hunk 12]
```diff
@@ -1,7 +1,105 @@
...
 
-class Labhub(BotPlugin):
-    """
-    GitHub and GitLab utilities
-    """
+from errbot import BotPlugin, re_botcmd
+
+
+class LabHub(BotPlugin):
+    """GitHub and GitLab utilities"""  # Ignore QuotesBear
```
Comment (sils) [inline plugins/labhub.py]: I'd just make this comment multiline

Comment (meetmangukiya) [inline plugins/labhub.py]: @jayvdb suggested me to PEP for this, so I think I'll keep it this way

**Comment by sils** [comment]
Commit message no verb and more than one thing in shortlog

File: plugins/labhub.py [hunk 13]
```diff
@@ -1,7 +1,105 @@
...
+
+    def activate(self):
+        super().activate()
+
+        with open(os.path.join(os.environ.get('COBOT_ROOT', os.getcwd()),
+                               'plugins/LabHub.json')) as f:
+            config = f.read()
+
+        config = json.loads(config)
```
Comment (sils) [inline plugins/labhub.py]: isn't this simpler?

```python
for key, value in config.items():
    setattr(self, key, value)
```

Or just store the values right in here,  I don't see how this json file is helping much

File: plugins/labhub.py [hunk 14]
```diff
@@ -1,7 +1,105 @@
...
+        return github3.login(token=token)
+
+    @staticmethod
+    def get_org(gh, org_name):
+        """
+        :param gh:          ``github3.GitHub`` object
+        :param org_name:    Name of the organization
+        :returns:           ``github3.orgs.Organization`` object
+        """
+        return gh.organization(org_name)
```
Comment (sils) [inline plugins/labhub.py]: this function is rather useless, a oneliner and only used once, I suggest inlining it

Comment (sils) [inline plugins/labhub.py]: this also applies to others

File: plugins/labhub.py [hunk 15]
```diff
@@ -1,7 +1,105 @@
...
+        return teams
+
+    @staticmethod
+    def invite(user, team):
+        """
+        Invite user to team.
+        :param user:    Username
+        :param team:    ``github3.orgs.Team`` object
+        """
+        return team.invite(user)
```
Comment (sils) [inline plugins/labhub.py]: again no benefit in adding a wrapper that just turns around the argument order

File: plugins/labhub.py (hunk 12 above)
Comment (sils) [inline plugins/labhub.py]: can you describe this better? Isn't this the invite feautre?

Comment (meetmangukiya) [inline plugins/labhub.py]: More things will come here, creating issues, assigning issues, all of it

File: plugins/labhub.py (hunk 13 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: this json makes the script configurable for others to adapt, without actually changing the script

**Comment by sils** [comment]
then add concrete examples please

**Comment by sils** [comment]
unack 7ed1aa0

**Comment by meetmangukiya** [comment]
NOTE for @Udayan12167 : Other then the `WIP: assign` HEAD commit everything else is ready, please review those first

File: plugins/labhub.py [hunk 16]
```diff
@@ -1,7 +1,101 @@
...
+ through https://github.com/coala/coala/wiki/Membership
+"""
+    }
+
+    ORG_USERNAME = constants.GH_ORG_NAME
+
+    @staticmethod
+    def get_teams(token, org_name):
+        """
+        Invite user to team.
```
Comment (Udayan12167) [inline plugins/labhub.py]: Incorrect doc string.

File: tests/labhub_test.py [hunk 17]
```diff
@@ -0,0 +1,77 @@
+import logging
+import os
+import unittest
+from unittest.mock import Mock, MagicMock, create_autospec, PropertyMock
+
+import github3
+import pytest
```
Comment (meetmangukiya) [inline tests/labhub_test.py]: remove this

File: tests/labhub_test.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+from errbot.backends.test import TestBot, FullStackTest
```
Comment (meetmangukiya) [inline tests/labhub_test.py]: remove `FullStackTest`

File: plugins/labhub.py [hunk 19]
```diff
@@ -34,7 +36,26 @@ class LabHub(BotPlugin):
...
+    GH_ORG_NAME = constants.GH_ORG_NAME
+    GL_ORG_NAME = constants.GL_ORG_NAME
+
+    def activate(self):
+        super().activate()
+        try:
+            self.IGH = GitHub(os.environ.get('GH_TOKEN'))
+            self.IGL = GitLab(os.environ.get('GL_TOKEN'))
+
+            self.gh_repos = {repo.full_name.split('/')[1]: repo for repo in
```
Comment (meetmangukiya) [inline plugins/labhub.py]: Add a comment ?

Comment (meetmangukiya) [inline plugins/labhub.py]: nah, not required

File: plugins/labhub.py [hunk 20]
```diff
@@ -90,12 +111,27 @@ def callback_message(self, msg):
         """Invite the user whose message includes the holy 'hello world'"""
         if not hasattr(self, 'TEAMS'):
             self.TEAMS = LabHub.get_teams(os.environ.get('GH_TOKEN'),
-                                          self.ORG_USERNAME)
+                                          self.GH_ORG_NAME)
```
Comment (Udayan12167) [inline plugins/labhub.py]: These are unrelated changes to the commit message? Split it into two. One with the constants change and one with the create issue feature added.

File: plugins/labhub.py [hunk 21]
```diff
@@ -90,12 +111,27 @@ def callback_message(self, msg):
...
+    def create_issut_cmd(self, msg, match):
+        repo_name = match.group(1)
+        iss_title = match.group(2)
+        iss_description = match.group(3) if match.group(3) is not None else ''
+        if repo_name in self.REPOS:
+            repo = self.REPOS[repo_name]
+            iss = repo.create_issue(iss_title, iss_description)
+            return 'Here you go: {}'.format(iss.url)
+        else:
+            return 'Can\'t create an issue for a repository that does not '\
```
Comment (Udayan12167) [inline plugins/labhub.py]: In brackets without the trailing `\`

File: plugins/labhub.py (hunk 20 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: or I'll just backport the same name to the first commit :+1:

File: plugins/labhub.py [hunk 22]
```diff
@@ -90,12 +111,27 @@ def callback_message(self, msg):
...
                           self.INVITE_SUCCESS['newcomers'].format(user))
-                self.TEAMS[self.ORG_USERNAME + ' newcomers'].invite(user)
+                self.TEAMS[self.GH_ORG_NAME + ' newcomers'].invite(user)
+
+    @re_botcmd(pattern=r'(?:new|file) issue ([\w-]+?)(?: |\n)(.+?)(?:$|\n((?:.|\n)*))',  # Ignore LineLengthBear, PyCodeStyleBear
+               flags=re.IGNORECASE)
+    def create_issut_cmd(self, msg, match):
+        repo_name = match.group(1)
+        iss_title = match.group(2)
+        iss_description = match.group(3) if match.group(3) is not None else ''
```
Comment (Udayan12167) [inline plugins/labhub.py]: What if the description is in line as well after a space. You match either terminator or new line but no space. Am I right in reading that regex.

Comment (meetmangukiya) [inline plugins/labhub.py]: if the description is made in line after a space, then there is no way to differentiate if that is part of the title or part of the description

File: plugins/labhub.py (hunk 20 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: I guess this is done, and also this looks misplaced, did you mean to add it on the second commit?

File: plugins/labhub.py [hunk 23]
```diff
@@ -1,7 +1,262 @@
...
+            # send the invite
+            self.TEAMS[team_mapping[team.lower()]].invite(invitee)
+            return self.INVITE_SUCCESS[team.lower()].format(invitee)
+        else:
+            return ('@{}, you are not a maintainer, only maintainers can invite'
+                    ' other people. Nice try :poop:'.format(inviter))
+
+    def callback_message(self, msg):
+        """Invite the user whose message includes the holy 'hello world'"""
+        if not hasattr(self, 'TEAMS'):
```
Comment (jayvdb) [inline plugins/labhub.py]: create a instance `@property` `teams`, which abstracts away the dynamic building of the data when required

File: plugins/labhub.py [hunk 24]
```diff
@@ -1,7 +1,262 @@
...
+            teams[team.name] = team
+        return teams
+
+    @re_botcmd(pattern=r'(?:(?:invite)|(?:inv))\s+(\w+)\s*(?:to)\s+(\w+)')
+    def invite_cmd(self, msg, match):
+        """
+        Invite given user to given team. By default it invites to
+        "newcomers" team.
+        """
+        if not hasattr(self, 'TEAMS'):
```
Comment (jayvdb) [inline plugins/labhub.py]: `hasattr(self, ...)` -> `@property`

File: plugins/labhub.py [hunk 25]
```diff
@@ -1,7 +1,258 @@
...
+    GL_ORG_NAME = constants.GL_ORG_NAME
+
+    def __init__(self, bot, name=None):
+        super().__init__(bot, name)
+
+        teams = dict()
+        try:
+            gh = github3.login(os.environ.get('GH_TOKEN'))
+            assert gh is not None
+            org = gh.organization(self.GH_ORG_NAME)
```
Comment (sadovnychyi) [inline plugins/labhub.py]: You *could* move this into an `else` clause, e.g.:

```python
try:
    gh = github3.login(os.environ.get('GH_TOKEN'))
    assert gh is not None
except AssertionError:
    self.log.error('Cannot create github object, please check GH_TOKEN')
else:
    org = gh.organization(self.GH_ORG_NAME)
    for team in org.iter_teams():
        teams[team.name] = team
```

From python [docs](https://docs.python.org/3/tutorial/errors.html#handling-exceptions):

> The use of the else clause is better than adding additional code to the try clause because it avoids accidentally catching an exception that wasn’t raised by the code being protected by the try ... except statement.

File: plugins/labhub.py [hunk 26]
```diff
@@ -1,7 +1,258 @@
...
+            for team in org.iter_teams():
+                teams[team.name] = team
+        except AssertionError:
+            self.log.error('Cannot create github object, please check GH_TOKEN')
+
+        self._teams = teams
+
+        try:
```
Comment (sadovnychyi) [inline plugins/labhub.py]: Same here, I would avoid putting a huge code chunk inside of a `try`.

File: plugins/labhub.py [hunk 27]
```diff
@@ -1,7 +1,258 @@
...
+            if not self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user):
+                # send the invite
+                self.send(msg.frm,
+                          self.INVITE_SUCCESS['newcomers'].format(user))
+                self.TEAMS[self.GH_ORG_NAME + ' newcomers'].invite(user)
+
+    @re_botcmd(pattern=r'(?:new|file) issue ([\w-]+?)(?: |\n)(.+?)(?:$|\n((?:.|\n)*))',  # Ignore LineLengthBear, PyCodeStyleBear
+               flags=re.IGNORECASE)
+    def create_issut_cmd(self, msg, match):
+        '''Create issues on GitHub and GitLab repositories.'''
```
Comment (sadovnychyi) [inline plugins/labhub.py]: Definitely not important but my perfectionism kills me seeing that you used double quotes for doc strings before but decided to use single quotes here. 🤢

Comment (meetmangukiya) [inline plugins/labhub.py]: yeah, this one was because QuotesBear is buggy for oneliner double quotes docstrings, and would have to add ignore comment everywhere. Thanks for pushing this, I'll add the ignores now :sweat_smile:

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/e9a6b23cbc3993630a0936543643870c85e661f3

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/413cbe40859d7227b19a17928cac313dbfcbe461

File: plugins/labhub.py [hunk 28]
```diff
@@ -136,3 +136,28 @@ def create_issut_cmd(self, msg, match):
...
+
+        user = msg.frm.nick
+
+        try:
+            assert org == self.GH_ORG_NAME or org == self.GL_ORG_NAME
+        except AssertionError:
+            return 'Repository not owned by our org.'
+
+        try:
+            iss = self.REPOS[repo_name].get_issue(int(issue_number))
```
Comment (Udayan12167) [inline plugins/labhub.py]: Can we apply the else fix here as well https://github.com/coala/corobo/pull/71/commits/e9a6b23cbc3993630a0936543643870c85e661f3#diff-dd3a6bcfd3fdd49afb75cd26d1429417R47

File: tests/helper.py [hunk 29]
```diff
@@ -0,0 +1,7 @@
+from errbot.backends.test import TestBot
```
Comment (Udayan12167) [inline tests/helper.py]: Can you separate this into another commit. This is unrelated to this commit.

File: requirements.txt [hunk 30]
```diff
@@ -1,3 +1,3 @@
 errbot
 github3.py
-IGitt==0.3.0.dev20170612081715
+IGitt==0.3.0.dev20170615131838
```
Comment (Udayan12167) [inline requirements.txt]: This is an unrelated change again.

File: tests/labhub_test.py [hunk 31]
```diff
@@ -72,6 +71,9 @@ def test_hello_world_callback(self):
         self.mock_team.invite.assert_called_with(None)
 
     def test_create_issue_cmd(self):
+        plugins.labhub.GitHub = create_autospec(IGitt.GitHub.GitHub.GitHub)
+        plugins.labhub.GitLab = create_autospec(IGitt.GitLab.GitLab.GitLab)
```
Comment (Udayan12167) [inline tests/labhub_test.py]: Similarly unrelated.

File: requirements.txt (hunk 30 above)
Comment (meetmangukiya) [inline requirements.txt]: this is a related change, the IGitt new functionality that are used in this commit are in this version

File: tests/helper.py (hunk 29 above)
Comment (meetmangukiya) [inline tests/helper.py]: yeah, seems right :+1:

File: plugins/labhub.py [hunk 32]
```diff
@@ -161,3 +161,41 @@ def unassign_cmd(self, msg, match):
...
+        """Mark a given PR/MR with status labels."""  # Ignore QuotesBear
+        state, host, org, repo_name, xr, number = match.groups()
+
+        if host.lower() == 'github':
+            assert xr.lower() == 'pull'
+        elif host.lower() == 'gitlab':
+            assert xr.lower() == 'merge_requests'
+
+        try:
+            mr = self.REPOS[repo_name].get_mr(number)
```
Comment (Udayan12167) [inline plugins/labhub.py]: Move everything else in the else clause. To not catch other exceptions.

File: plugins/labhub.py [hunk 33]
```diff
@@ -161,3 +161,41 @@ def unassign_cmd(self, msg, match):
...
+            current_labels = mr.labels
+            if state == 'wip':
+                pending_labels = ['process/pending_review',
+                                  'process/pending review']
+                for label in pending_labels:
+                    current_labels.remove(label)
```
Comment (Udayan12167) [inline plugins/labhub.py]: Is this a list. This will throw an error if there is no such item.

File: tests/labhub_test.py (hunk 31 above)
Comment (meetmangukiya) [inline tests/labhub_test.py]: not unrelated, I don't want these in the test_mark_cmd, hence moved from setUp to tests where they were required

File: plugins/labhub.py (hunk 33 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: yep, that is a list, valid concern, nice catch! :+1:

Comment (Udayan12167) [inline plugins/labhub.py]: You can do it like `[current_labels.remove(label) for lable in pending_labels if label in current_labels]`

Comment (meetmangukiya) [inline plugins/labhub.py]: but that'll return an empty list, and I don't think we really need a comprehension here :confused:

Comment (meetmangukiya) [inline plugins/labhub.py]: lets, use a `filter` instead

File: plugins/labhub.py [hunk 34]
```diff
@@ -201,3 +201,62 @@ def mark_cmd(self, msg, match):
...
+
+        user = msg.frm.nick
+
+        try:
+            assert org == self.GH_ORG_NAME or org == self.GL_ORG_NAME
+        except AssertionError:
+            yield 'Repository not owned by our org.'
+            return
+
+        checks = [
```
Comment (Udayan12167) [inline plugins/labhub.py]: I would rather you just create a separate method. If you want to add a method to a list to use the for loop in eligible use a decorator to do so.
```
checks = []


def register_check(function):
    checks.append(function)

    return function

@register_check
def function_to_add_to_check():
    """
    I would like a docstring for the check.
    """
```

Comment (Udayan12167) [inline plugins/labhub.py]: You can move this out of the class as well.

Comment (meetmangukiya) [inline plugins/labhub.py]: aren't lambdas better for this kind of things :confused:

Comment (meetmangukiya) [inline plugins/labhub.py]: this is more readable though, doing this :+1:

File: plugins/labhub.py (hunk 28 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: done

File: tests/helper.py (hunk 29 above)
Comment (meetmangukiya) [inline tests/helper.py]: done

File: requirements.txt (hunk 30 above)
Comment (meetmangukiya) [inline requirements.txt]: invalid

File: tests/labhub_test.py (hunk 31 above)
Comment (meetmangukiya) [inline tests/labhub_test.py]: invalid

File: plugins/labhub.py (hunk 32 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: done

File: plugins/labhub.py (hunk 33 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: done

Comment (meetmangukiya) [inline plugins/labhub.py]: done

File: plugins/labhub.py (hunk 34 above)
Comment (meetmangukiya) [inline plugins/labhub.py]: done, no, let it be in the class IMO

File: tests/helper.py (hunk 29 above)
Comment (meetmangukiya) [inline tests/helper.py]: seems like I messed this one up

Comment (meetmangukiya) [inline tests/helper.py]: done, now :+1:

**Comment by meetmangukiya** [comment]
seems like there is some coverage issue, fixing ASAP

File: plugins/labhub.py [hunk 35]
```diff
@@ -204,3 +204,76 @@ def mark_cmd(self, msg, match):
...
+               than low.
+            """
+            if self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user):
+                diff_labels = filter(lambda x: 'difficulty' in x, iss.labels)
+                if list(filter(lambda x: ('low' in x) or ('newcomer' in x),
+                               diff_labels)):
+                    return True
+                else:
+                    return False
+            else:
```
Comment (Udayan12167) [inline plugins/labhub.py]: Umm wouldn't this condition allow a user who is not even a newcomer to assign to issues? Like doesn't belong to any team?

Comment (meetmangukiya) [inline plugins/labhub.py]: hmm.. yes, but in that case assign will fail automatically, you cannot assign an issue that is not part of the org

Comment (Udayan12167) [inline plugins/labhub.py]: Oh okay cool.

File: plugins/labhub.py [hunk 36]
```diff
@@ -204,3 +204,76 @@ def mark_cmd(self, msg, match):
...
+        ]
+
+        try:
+            iss = self.REPOS[repo_name].get_issue(int(iss_number))
+        except KeyError:
+            yield 'Repository doesn\'t exist.'
+        else:
+            if not iss.assignees:
+                if eligible(user, iss):
+                    iss.assign(user)
```
Comment (Udayan12167) [inline plugins/labhub.py]: If assign fails will there be an exception?

Comment (meetmangukiya) [inline plugins/labhub.py]: I guess, yes, there'll be, fixing it :rocket:

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/5d19de212df0af6994145dc7f4a35fc2496f2b50

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/99c6056957d84a4486ad9e00e17ab3d43f9cd9e0

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/dcfb0306960afa0cfb9c9a0ecc4bab9331a2aef9

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/236ef6e48ace74adf3d33de2612d7dc3f7f5f88c

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/4eb50c6937b803aa63529143a1375b21fe2ee13b

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/71/commits/8d1d3d2791b3d1d236ccf7ad0406d9566bf5b3da

**Comment by meetmangukiya** [comment]
ack c83e7678dc7feb156778fc78eb8c33fef1f2aa25

**Comment by sils** [comment]
ack c83e767

**Comment by hemangsk** [comment]
ack ce2a27a

**Comment by hemangsk** [comment]
ack c83e7678dc7feb156778fc78eb8c33fef1f2aa25

**Comment by arjunsinghy96** [comment]
ack ce2a27a

**Comment by arjunsinghy96** [comment]
ack ce2a27a

**Comment by arjunsinghy96** [comment]
ack c83e767

**Comment by arjunsinghy96** [comment]
4ae86a7 ack

**Comment by arjunsinghy96** [comment]
ack 5271c55

**Comment by arjunsinghy96** [comment]
ack b4895f9

**Comment by arjunsinghy96** [comment]
b4895f9

**Comment by meetmangukiya** [comment]
ack a7c8cb10d55bca717e6ac3a48969878a75b3adb0

**Comment by meetmangukiya** [comment]
ack 68f4b85702bc6fc5b3cecd278382a5de0cf1d78e

**Comment by meetmangukiya** [comment]
ack 93b5b802c72e85ee02f2d2c9b6e1163ea5eed51b

**Comment by meetmangukiya** [comment]
ack e541c60a2b0609e3311bd8ed6234034534f9b967

**Comment by meetmangukiya** [comment]
@rultor merge
